packages/components/menu/src/menu-group.tsx

```diff
@@ -1,24 +1,68 @@
 import { HTMLChakraProps, chakra, forwardRef } from "@chakra-ui/system"
 import { cx } from "@chakra-ui/shared-utils"
+import { useVirtualizer } from "@tanstack/react-virtual"
 
 import { useMenuStyles } from "./menu"
+import { useRef } from "react"
 
-export interface MenuGroupProps extends HTMLChakraProps<"div"> {}
+export interface MenuGroupProps extends HTMLChakraProps<"div"> {
+  virtualized?: boolean
+  virtualizedOptions?: {
+    rowHeight: number
+    containerHeight: number
+  }
+}
 
 export const MenuGroup = forwardRef<MenuGroupProps, "div">((props, ref) => {
   const { title, children, className, ...rest } = props
+  const arrayChildren = children as any[] // FIX this
 
+  const virtualizedRef = useRef(null)
   const _className = cx("chakra-menu__group__title", className)
   const styles = useMenuStyles()
 
+  const rowVirtualizer = useVirtualizer({
+    count: arrayChildren?.length,
+    getScrollElement: () => virtualizedRef.current,
+    estimateSize: () => props.virtualizedOptions?.rowHeight ?? 35,
```

Comment:
Should we fetch the default `rowHeight` and `containerHeight` values from a hb-token, rather than putting a hard number here?